bin/onlyswaps-verifier/src/evaluator.rs

```diff
@@ -0,0 +1,295 @@
+use crate::chain_state_resolver::ChainState;
+use alloy::primitives::U256;
+use anyhow::anyhow;
+use chrono::{DateTime, TimeDelta, Utc};
+use config::network::NetworkConfig;
+use std::collections::HashMap;
+use std::ops::Add;
+
+type ChainId = U256;
+#[derive(Clone)]
+pub struct Evaluator {
+    finality_durations: HashMap<ChainId, TimeDelta>,
+}
+impl Evaluator {
+    pub fn new(finalisation_durations: &[NetworkConfig]) -> anyhow::Result<Self> {
+        let mut finality_durations = HashMap::new();
+
+        for config in finalisation_durations {
+            finality_durations.insert(
+                U256::from(config.chain_id),
+                TimeDelta::from_std(config.finality_duration_secs)?,
+            );
+        }
+
+        Ok(Self { finality_durations })
+    }
+    pub fn evaluate(&self, chain_state: ChainState) -> anyhow::Result<ChainState> {
+        let ChainState {
+            swap_params,
+            transfer_receipt,
+        } = &chain_state;
+        if transfer_receipt.dstChainId != swap_params.dstChainId {
+            anyhow::bail!("funds were sent on the wrong chain")
+        }
+
+        // this shouldn't be possible
+        if transfer_receipt.srcChainId != swap_params.srcChainId {
+            anyhow::bail!("funds were sent from the wrong chain")
+        }
+
+        if transfer_receipt.recipient != swap_params.recipient {
+            anyhow::bail!("funds were sent to the wrong recipient")
+        }
+
+        // right now this relies on the fact that tokens have the same address
+        // on multiple chains which could be a bug
```

Comment:
pretty sure that comment is a left over from before we had both `tokenIn` and `tokenOut`, but since now we have these it's no true anymore, no?